embulk-core/src/main/java/org/embulk/plugin/maven/MavenArtifactFinder.java

```diff
@@ -0,0 +1,122 @@
+package org.embulk.plugin.maven;
+
+import java.io.IOError;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.NotDirectoryException;
+import java.nio.file.Path;
+import org.apache.maven.repository.internal.MavenRepositorySystemUtils;
+import org.eclipse.aether.DefaultRepositorySystemSession;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.impl.DefaultServiceLocator;
+import org.eclipse.aether.repository.LocalRepository;
+import org.eclipse.aether.resolution.ArtifactRequest;
+import org.eclipse.aether.resolution.ArtifactResolutionException;
+import org.eclipse.aether.resolution.ArtifactResult;
+
+public class MavenArtifactFinder
+{
+    private MavenArtifactFinder(final Path givenLocalMavenRepositoryPath,
+                                final Path absoluteLocalMavenRepositoryPath,
+                                final RepositorySystem repositorySystem,
+                                final RepositorySystemSession repositorySystemSession)
+    {
+        this.givenLocalMavenRepositoryPath = givenLocalMavenRepositoryPath;
+        this.absoluteLocalMavenRepositoryPath = absoluteLocalMavenRepositoryPath;
+        this.repositorySystem = repositorySystem;
+        this.repositorySystemSession = repositorySystemSession;
+    }
+
+    public static MavenArtifactFinder create(final Path localMavenRepositoryPath)
+        throws MavenRepositoryNotFoundException
+    {
+        final Path absolutePath;
+        try {
+            absolutePath = localMavenRepositoryPath.normalize().toAbsolutePath();
+        }
+        catch (IOError ex) {
+            throw new MavenRepositoryNotFoundException(localMavenRepositoryPath, ex);
+        }
+        catch (SecurityException ex) {
+            throw new MavenRepositoryNotFoundException(localMavenRepositoryPath, ex);
+        }
```

Comment:
minor: Cannot we catch handle both of `IOExrror` and `SecurityException` in a single catch statement? like `catch (IOError | SecurityException) {`